src/views/Home.vue

```diff
@@ -199,15 +191,32 @@ export default {
       return response.ok ? response.json() : [];
     },
     getPlanDataFromUrl() {
-      const path = window.location.hash;
+      let path = window.location.hash;
       const planIndicator = '#/plan/';
       const moduleSeparator = '_';
       const semesterSeparator = '-';
       function isNullOrWhitespace(input) {
         return !input || !input.trim();
       }
+
+      if (!path.startsWith(planIndicator)) {
+        const cachedPlan = localStorage.getItem('plan');
+        if (cachedPlan) {
+          window.location.hash = cachedPlan;
+          path = cachedPlan;
+        }
+      }
+
       if (path.startsWith(planIndicator)) {
-        return path
+        // This ensures backwards compatability. Removing it after everyone who started before 2022
```

Comment:
@StefanieJaeger I've allowed myself to push this commit - I have two reasons for that:

My plan (in my cache) caused an error, so I added SEP1 and SEP2 migrations.

Additionally, I want to be able to check my plans in the next two years without manually starting an old version - and I have them in my saved messages on telegram (so local storage is not sufficient) where I copy them from. They still contain SE1 and 2, so I kind of don't want that support removed right now.

Is that okay?

Reply:
I'm sure I'm not the only one :)

Reply:
I thought we already had this somewhere, but I can't find it.

Reply:
Btw @Mafii I'd prefer if you could use GitHub's "Suggest Changes" feature instead of pushing directly in the future. This way the PR author keeps the control.

Reply:
I would have also preferred a code suggestion for me to accept.
But the code is fine like this.

Reply:
sorry, will do next time!